### ISSUE TITLE
[FutureWarning] Conversion from `float` to `np.floating` is deprecated

### ISSUE THREAD

**Issue by Androbin** [h5py/h5py#1033]
* Operating System: `Ubuntu 16.04.2 LTS`
 * Python version: `3.5.2`
 * Where Python was acquired: `system Python`
 * h5py version: `2.7.1`
 * HDF5 version: `1.8.18`

The full traceback/stack trace shown:
```
/usr/local/lib/python3.5/dist-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
```

**Comment by pharshalp** [comment]
I think this has already been fixed by #970

**Comment by tacaswell** [comment]
You can get the rc with this fix via `pip install --upgrade --pre h5py`
